library/core/src/convert/num.rs

```diff
@@ -45,7 +45,8 @@ impl_float_to_int!(f64 => u8 u16 u32 u64 u128 usize i8 i16 i32 i64 i128 isize);
 macro_rules! impl_from {
     ($Small: ty, $Large: ty, #[$attr:meta], $doc: expr) => {
         #[$attr]
-        impl From<$Small> for $Large {
+        #[rustc_const_unstable(feature = "const_num_from_num", issue = "87852")]
```

Comment:
Stability attributes do not usually work on trait `impl`s. Do they work for `impl const`?

Reply:
Yes. See [this](https://github.com/rust-lang/rust/blob/master/src/test/ui/rfc-2632-const-trait-impl/staged-api.rs).

Reply:
That tests the case where we call a trait fn on a concrete type.

But there is another way to use a trait impl: to satisfy a bound.
```rust
const fn const_context_generic<T: MyTrait>() {
  T::func();
}

const_context_generic::<Unstable>();
```
Is that properly checked?

Reply:
That tests the case where we call a trait fn on a concrete type.

But there is another way to use a trait impl: to satisfy a bound.
```rust
const fn const_context_generic<T: MyTrait>() {
  T::func();
}

const_context_generic::<Unstable>();
```
Is that properly checked?

Reply:
You are right, those are probably not. I guess it is fine to let this PR pass and I will come up with a fix in the future. (and we need to wait for #87375, as that introduces a check on const bounds to error on previous code that should not be accepted) Trait bounds on const functons are gated behind a feature anyways, so even if this would allow one to use an unstable library feature without having it enabled, it would still require users to use a nightly compiler anyways.